### PR TITLE
Add taxonomy command

### PR DESCRIPTION
**Pull request by antgonza**
Depends on https://github.com/qiita-spots/qtp-diversity/pull/11; so first review/merge that one.

### CODE REVIEW

**Review by charles-cowart — APPROVED** [review]

File: qp_qiime2/__init__.py [hunk 1]
```diff
@@ -15,24 +18,40 @@
...
+# The extra commands require a folder where all the pre-calculated databases
+# exist, which is set up via a ENV variable, if not present we should raise
+# an error
+qp_qiime2_dbs = environ.get('QP_QIIME2_DBS')
+if qp_qiime2_dbs is None:
+    raise ValueError("Missing ENV var QP_QIIME2_DBS, please set.")
+qp_qiime2_dbs = glob(join(qp_qiime2_dbs, '*.qza'))
+if len(qp_qiime2_dbs) < 1:
+    raise ValueError(
+        "ENV QP_QIIME2_DBS points to a folder without qza's, please set.")
```
Comment (unique-identifier) [inline qp_qiime2/__init__.py]: I believe qza's -> QZAs or QZA files.

File: qp_qiime2/__init__.py [hunk 2]
```diff
@@ -228,3 +243,71 @@
...
+        req_params[element.description] = ('artifact', ['BIOM'])
+    elif eqt == 'TaxonomicClassifier':
+        default = qp_qiime2_dbs[0]
+        qp_qiime2_dbs = ', '.join('"%s"' % db for db in qp_qiime2_dbs)
+
+        ename = '%s (%s)' % (element.description, pname)
+        req_params[ename] = ('choice:[%s]' % qp_qiime2_dbs, default)
+        req_params['qp-hide-param' + ename] = ('string', pname)
+    else:
+        raise ValueError('Found non expected input: "%s", in '
```
Comment (unique-identifier) [inline qp_qiime2/__init__.py]: non expected -> unexpected

File: qp_qiime2/tests/test_qiime2.py [hunk 3]
```diff
@@ -76,6 +79,114 @@ def test_not_analysis_artifact(self):
...
 
+    def test_feature_classifier(self):
+        dbpath = join(self.basedir, '..', '..', 'databases',
+                      'gg-13-8-99-515-806-nb-classifier.qza')
+        original_params = {
+            'qp-hide-method': 'classify_sklearn',
+            'qp-hide-plugin': 'feature-classifier',
+            'The feature data to be classified.': '8',
+            'qp-hide-paramDirection of reads with respect to reference '
+            'sequences. same will cause reads to be classified unchanged; '
```
Comment (unique-identifier) [inline qp_qiime2/tests/test_qiime2.py:90]: same -> Same?

File: qp_qiime2/tests/test_qiime2.py [hunk 4]
```diff
@@ -76,6 +79,114 @@ def test_not_analysis_artifact(self):
...
+            'processes. If -1 all CPUs are used. If 1 is given, no parallel '
+            'computing code is used at all, which is useful for debugging. '
+            'For n_jobs below -1, (n_cpus + 1 + n_jobs) are used. Thus for '
+            'n_jobs = -2, all CPUs but one are used. (n_jobs)': 'n_jobs',
+            'The maximum number of concurrently worker processes. If -1 all '
+            'CPUs are used. If 1 is given, no parallel computing code is '
+            'used at all, which is useful for debugging. For n_jobs below -1, '
+            '(n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs '
+            'but one are used. (n_jobs)': '1', 'qp-hide-paramNumber of reads '
+            'to process in each batch. If 0, this parameter is autoscaled to '
```
Comment (unique-identifier) [inline qp_qiime2/tests/test_qiime2.py:122]: autoscaled -> auto-scaled?

**Review by unique-identifier — CHANGES_REQUESTED** [review]

**Review by unique-identifier — APPROVED** [review]

File: qp_qiime2/__init__.py (hunk 1 above)
Comment (antgonza) [reply qp_qiime2/__init__.py]: K, thanks!

File: qp_qiime2/__init__.py (hunk 2 above)
Comment (antgonza) [reply qp_qiime2/__init__.py]: k, thanks.

File: qp_qiime2/tests/test_qiime2.py (hunk 3 above)
Comment (antgonza) [reply qp_qiime2/tests/test_qiime2.py:90]: This comes from Q2, if you feel strongly about it, please open an issue directly in the plugin feature-classifier.

File: qp_qiime2/tests/test_qiime2.py (hunk 4 above)
Comment (antgonza) [reply qp_qiime2/tests/test_qiime2.py:122]: This comes from Q2, if you feel strongly about it, please open an issue directly in the plugin feature-classifier.

**Review by antgonza — COMMENTED** [review]
Thanks for review!

**Comment by antgonza** [comment]
Thanks for review. Yes, we used to check for failure but as we changed the mapping file in Qiita now the command is actually successful so we changed the test.

**Pull request merged**